pkg/dns/service.go

```diff
@@ -79,13 +79,25 @@ func (s *Service) GetDNSRecords(ctx context.Context, traffic traffic.Interface)
 		record := &v1alpha1.DNSRecord{
 			ObjectMeta: metav1.ObjectMeta{
 				Name:      host,
-				Namespace: s.defaultCtrlNS,
+				Namespace: traffic.GetNamespace(),
 			},
 		}
 		if err := s.controlClient.Get(ctx, client.ObjectKeyFromObject(record), record); err != nil {
 			if k8serrors.IsNotFound(err) {
-				log.Log.V(10).Info("no dnsrecord found for host ", "host", record.Name)
-				continue
+				log.Log.V(10).Info("no dnsrecord found for host ", "host", record.Name, "namespace", record.Namespace)
```

Comment:
@mikenairn I've updated this to check the tenant namespace first for a matching MZ, then fallback to the default ns

Reply:
Hmm, do we know what managed zone is being used by this point? It should only be looking in that ns i think since the record will always be in the MZ ns.

Reply:
but it can't check the defaultNS that's for sure

Reply:
GetDNSRecords is called from 3 places:
- EnsureManagedHost (ingress specific)
- AddEndPoints
- RemoveEndpoints

RemoveEndpoints is not used anywhere yet, but I'd imagine you'd call it when a Gateway (or Ingress) is deleted, or maybe when a specific listener block is removed from a Gateway.
In those cases the MZ would have to be looked up from the host. That being said, there's an edge case where an MZ could be added after the fact that matches the host, but the DNSRecord was created against a different MZ.

AddEndPoints isn't used by Ingress, as there's patching that handles that.
For Gateways, the MZ is known. If we were to pass it in to GetDNSRecords, the logic in the gateway reconciler would need to be refactored somewhat to map hosts to MZs early on and keep that map around until AddEndPoints has to be called for each host.

I'm not sure about EnsureManagedHost.

Reply:
@mikenairn There's a typo in my first comment here.
~check the tenant namespace first for a matching MZ~
should be 'check the tenant namespace first for a matching DNSRecord by host'

Reply:
I discovered a bug here where it returns too early instead of adding the DNSRecord to the array. I've pushed a small change https://github.com/Kuadrant/multi-cluster-traffic-controller/pull/84/commits/748dcd0aa5466d102ab3e1294c971308029983bb

Reply:
I created a follow on issue for this https://github.com/Kuadrant/multi-cluster-traffic-controller/issues/111